lib/bots/versionbot.ts

```diff
@@ -390,6 +409,228 @@ export class VersionBot extends ProcBot {
+     *
+     * @param _registration GithubRegistration object used to register the method
+     * @param event         ServiceEvent containing the event information ('pull_request' event)
+     * @returns             A void Promise once execution has finished.
+     */
+    protected addReviewers = (_registration: GithubRegistration, event: ServiceEvent): Promise<void> => {
+        const pr = event.cookedEvent.data.pull_request;
+        const head = event.cookedEvent.data.pull_request.head;
+        const owner = head.repo.owner.login;
+        const repo = head.repo.name;
+        let approvedMaintainers: string[];
+        let approvedReviewers: string[];
+
+        // Only when opening a new PR.
+        console.log(event.cookedEvent.data.action);
+        if (event.cookedEvent.data.action !== 'opened') {
+            return Promise.resolve();
+        }
+
+        this.logger.log(LogLevel.INFO, `Checking reviewers list for ${owner}/${repo}#${pr.number}`);
+
+        // Get the reviewers for the PR.
+        return this.retrieveConfiguration({
+            emitter: this.githubEmitter,
+            location: {
+                owner,
+                repo,
+                path: RepositoryFilePath
+            }
+        }).then((config: VersionBotConfiguration) => {
+            approvedMaintainers = this.stripPRAuthor((config || {}).maintainers || null, pr) || [];
+            approvedReviewers = this.stripPRAuthor((config || {}).reviewers || null, pr) || [];
+
+            return this.dispatchToEmitter(this.githubEmitterName, {
+                data: {
+                    number: pr.number,
+                    owner,
+                    repo
+                },
+                method: this.githubApi.pullRequests.get
+            });
+        }).then((pullRequest: GithubApiTypes.PullRequest) => {
+            // Look at the reviewers, create a list of configured reviewers that were not added
+            // when the PR was opened.
+            const assignedReviewers = _.map(pullRequest.requested_reviewers, (reviewer) => reviewer.login);
+            const configuredReviewers = _.uniq(_.unionWith(approvedReviewers, approvedMaintainers));
+            let missingReviewers = _.filter(configuredReviewers, (reviewer) => {
+                return !(_.find(assignedReviewers, (assignedReviewer) => (assignedReviewer === reviewer)) ||
+                    (reviewer === pr.user.login));
+            });
+
+            // We don't assign the author as a reviewer, if they're in the list.
+            // An App has no ability to create a review request. The best we can do is to create a new
+            // comment pinging those missing in the list to add themselves.
+            if (missingReviewers.length > 0) {
+                let reviewerMessage = '';
+                _.each(missingReviewers, (reviewer) => {
+                    reviewerMessage += `@${reviewer}, `;
```

Comment:
But will it actually ping the person with two `@@` at the front is what I'm worried about. I guess we can test right now :) @@heds